pkg/domain/infra/tunnel/containers.go

```diff
@@ -0,0 +1,42 @@
+package tunnel
+
+import (
+	"context"
+
+	"github.com/containers/libpod/pkg/bindings/containers"
+	"github.com/containers/libpod/pkg/domain/entities"
+)
+
+func (ic *ContainerEngine) ContainerExists(ctx context.Context, nameOrId string) (*entities.BoolReport, error) {
+	exists, err := containers.Exists(ic.ClientCxt, nameOrId)
+	return &entities.BoolReport{Value: exists}, err
+}
+
+func (ic *ContainerEngine) ContainerWait(ctx context.Context, namesOrIds []string, options entities.WaitOptions) ([]entities.WaitReport, error) {
+	var (
+		responses []entities.WaitReport
+	)
+	cons, err := getContainersByContext(ic.ClientCxt, false, namesOrIds)
```

Comment:
I am undecided, whether it would be better to pass the whole ContainerEngine vs. just the connection context.

Reply:
sure, my assumption is this helper is likely a WIP